src/server/scripts/EasternKingdoms/SunwellPlateau/boss_muru.cpp

```diff
@@ -175,123 +140,72 @@ struct boss_entropius : public ScriptedAI
-                me->CastSpell(target, SPELL_BLACK_HOLE, true);
-            events.ScheduleEvent(EVENT_SPAWN_BLACK_HOLE, 15000);
-            break;
-        }
-
-        DoMeleeAttackIfReady();
+        scheduler.Update(diff);
     }
 };
 
 struct npc_singularity : public NullCreatureAI
 {
-    npc_singularity(Creature* creature) : NullCreatureAI(creature)
-    {
-    }
-
-    EventMap events;
+    npc_singularity(Creature* creature) : NullCreatureAI(creature) { }
 
     void Reset() override
     {
         me->DespawnOrUnsummon(18000);
-        me->CastSpell(me, SPELL_BLACK_HOLE_SUMMON_VISUAL, true);
-        me->CastSpell(me, SPELL_BLACK_HOLE_SUMMON_VISUAL2, true);
-        events.ScheduleEvent(EVENT_START_BLACK_HOLE, 3500);
-        events.ScheduleEvent(EVENT_SWITCH_BLACK_HOLE_TARGET, 5000);
-        events.ScheduleEvent(EVENT_SINGULARITY_DEATH, 17000);
-    }
+        DoCastSelf(SPELL_BLACK_HOLE_SUMMON_VISUAL, true);
+        DoCastSelf(SPELL_BLACK_HOLE_SUMMON_VISUAL2, true);
 
-    void UpdateAI(uint32 diff) override
-    {
-        events.Update(diff);
-        switch (events.ExecuteEvent())
-        {
-        case EVENT_SINGULARITY_DEATH:
+        me->m_Events.AddEventAtOffset([&] {
             me->KillSelf();
-            break;
-        case EVENT_START_BLACK_HOLE:
+        }, 17s);
+
+        me->m_Events.AddEventAtOffset([&] {
             me->RemoveAurasDueToSpell(SPELL_BLACK_HOLE_SUMMON_VISUAL2);
-            me->CastSpell(me, SPELL_BLACK_HOLE_VISUAL2, true);
-            me->CastSpell(me, SPELL_BLACK_HOLE_PASSIVE, true);
-            break;
-        case EVENT_SWITCH_BLACK_HOLE_TARGET:
+            DoCastSelf(SPELL_BLACK_HOLE_VISUAL2, true);
+            DoCastSelf(SPELL_BLACK_HOLE_PASSIVE, true);
+        }, 3500ms);
+
+        scheduler.Schedule(5s, [this](TaskContext context)
         {
-            Map::PlayerList const& players = me->GetMap()->GetPlayers();
-            for (Map::PlayerList::const_iterator itr = players.begin(); itr != players.end(); ++itr)
-                if (Player* player = itr->GetSource())
+            auto const& playerList = me->GetMap()->GetPlayers();
+            for (auto const& playerRef : playerList)
+            {
+                if (Player* player = playerRef.GetSource())
                     if (me->GetDistance2d(player) < 15.0f && player->GetPositionZ() < 72.0f && player->IsAlive() && !player->HasAura(SPELL_BLACK_HOLE_EFFECT))
                     {
                         me->GetMotionMaster()->MovePoint(0, player->GetPositionX(), player->GetPositionY(), player->GetPositionZ(), false, true);
-                        events.ScheduleEvent(EVENT_SWITCH_BLACK_HOLE_TARGET, 5000);
+                        context.Repeat();
                         return;
                     }
-            events.ScheduleEvent(EVENT_SWITCH_BLACK_HOLE_TARGET, 500);
-            break;
-        }
-        }
+            }
+
+            context.Repeat(1s);
```

Comment:
This was 500ms

Reply:
Yes, and I changed it to 1s. Because for no reason we would ever need such a low timer.